raft/src/main/java/org/apache/kafka/raft/RaftConfig.java

```diff
@@ -102,31 +120,23 @@ public boolean equals(Object obj) {
                 return false;
             }
 
-            final AddressSpec that = (AddressSpec) obj;
-            return that.address().equals(address());
+            final InetAddressSpec that = (InetAddressSpec) obj;
+            return that.address.equals(address);
         }
     }
 
-    public static class InetAddressSpec extends AddressSpec {
-        private final InetSocketAddress address;
-
-        public InetAddressSpec(InetSocketAddress address) {
-            if (address.equals(UNROUTABLE_ADDRESS)) {
-                throw new IllegalArgumentException("Address not routable");
-            }
-            this.address = address;
+    public static class UnknownAddressSpec implements AddressSpec {
+        private UnknownAddressSpec() {
```

Comment:
Since we have a private constructor, I think we will only have the `UNKNOWN_ADDRESS_SPEC_INSTANCE` instance. So can we rely on the default equals/hashcode?

Reply:
Yea, that's true actually. Should work fine for the map equals test case containing an instance of the UnknownAddressSpec. Removed